invenio_app_ils/config.py

```diff
@@ -1065,18 +1091,58 @@ def _(x):
                 terms=dict(field="internal_location.name")
             ),
         ),
+        filters=dict(
+            circulation=custom_exists_filter("circulation.state",
+                                             "NOT_ON_LOAN"),
+        ),
         post_filters=dict(
             status=terms_filter("status"),
             medium=terms_filter("medium"),
-            circulation=terms_filter("circulation.state"),
             restrictions=terms_filter("circulation_restriction"),
             location=terms_filter("internal_location.location.name"),
             internal_location=terms_filter("internal_location.name"),
         ),
     ),
     loans=dict(  # IlsLoansSearch.Meta.index
-        aggs=dict(state=dict(terms=dict(field="state"))),
-        post_filters=dict(state=terms_filter("state")),
+        aggs=dict(
+            state=dict(terms=dict(field="state")),
+            delivery=dict(terms=dict(field="delivery.method")),
+            returns=dict(
+                filter=dict(term=dict(state="ITEM_ON_LOAN")),
+                aggs=dict(
+                    end_date=dict(
+                        range=dict(
+                            field="end_date",
+                            ranges=[{"key": "Overdue",
+                                     "to": str(
+                                         (arrow.utcnow()).date())},
```

Comment:
Oh wow, now I understand. I am not sure this is evaluated when you perform the query... can you confirm that?
It probably is because `invenio-records-rest` gets the facet configuration on every request... but better double check it!